variables.tf

```diff
@@ -378,6 +378,12 @@ variable "subnet_creation_enabled" {
   nullable    = false
 }
 
+variable "subscription_id" {
```

Comment:
With azapi modules I think we should standardise on using parent_id as an input variable. We could then use [`provider::azapi::parse_resource_id`](https://registry.terraform.io/providers/Azure/azapi/latest/docs/functions/parse_resource_id) to get the details. WDYT?